tables/leaf.py

```diff
@@ -316,7 +316,7 @@ def _calc_chunkshape(self, expectedrows, rowsize, itemsize):
 
         # Compute the chunksize
         MB = 1024 * 1024
-        expected_mb = (expectedrows * rowsize) // MB
+        expected_mb = np.int32((np.int64(expectedrows) * rowsize) // MB)
```

Comment:
Sorry, I have not analysed the problem in depth but the need of a double casting looks very strange to me.
What is the data type of `expectedrows` normally?
In any case Python `int` should never be subject to overflow. Is it not enough ro have a single cast to `int`?

Reply:
@wony-zheng could you please confirm that the following path works for you?

```patch
-        expected_mb = (expectedrows * rowsize) // MB
+        expected_mb = (int(expectedrows) * rowsize) // MB)
```